src/toil/batchSystems/slurm.py

```diff
@@ -218,32 +222,90 @@ def sbatch(self, sbatch_line):
         except OSError as e:
             logger.error("sbatch command failed")
             raise e
-
+    
     def getJobExitCode(self, slurmJobID):
         logger.debug("Getting exit code for slurm job %d", slurmJobID)
+        
+        state, rc = self._getJobDetailsFromSacct(slurmJobID)
+        
+        if rc == -999:
+            state, rc = self._getJobDetailsFromScontrol(slurmJobID)
+
+        logger.debug("s job state is %s", state)
+        # If Job is in a running state, return None to indicate we don't have an update                                 
+        if state in ('PENDING', 'RUNNING', 'CONFIGURING', 'COMPLETING', 'RESIZING', 'SUSPENDED'):
+            return None
+
+        return rc
+
+    def _getJobDetailsFromSacct(self, slurmJobID):
         # SLURM job exit codes are obtained by running sacct.
         args = ['sacct',
                 '-n', # no header
                 '-j', str(slurmJobID), # job
                 '--format', 'State,ExitCode', # specify output columns
                 '-P', # separate columns with pipes
                 '-S', '1970-01-01'] # override start time limit
+
         process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        rc = process.returncode
+
+        if rc != 0:
+            # no accounting system or some other error
+            return (None, -999)
+
         for line in process.stdout:
             values = line.strip().split('|')
             if len(values) < 2:
                 continue
             state, exitcode = values
             logger.debug("sacct job state is %s", state)
             # If Job is in a running state, return None to indicate we don't have an update
-            if state in ('PENDING', 'RUNNING', 'CONFIGURING', 'COMPLETING', 'RESIZING', 'SUSPENDED'):
-                return None
             status, _ = exitcode.split(':')
             logger.debug("sacct exit code is %s, returning status %s", exitcode, status)
-            return int(status)
+            return (state, int(status))
+        
         logger.debug("Did not find exit code for job in sacct output")
-        return None
+        return (None, None)
 
+    def _getJobDetailsFromScontrol(self, slurmJobID):
+        args = ['scontrol',
+                'show',
+                'job',
+                str(slurmJobID)]
+
+        process = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+        job = dict()
+        for line in process.stdout:
+            values = line.strip().split()
+
+            # If job information is not available an error is issued:
+            # slurm_load_jobs error: Invalid job id specified
+            # There is job information, so exit.
```

Comment:
If I'm understanding this right, shouldn't this be: `There is no job information, so exit`?